.github/workflows/test-dbt-installation-source.yml

```diff
@@ -32,10 +38,55 @@ jobs:
   source-integration-test:
     runs-on: ubuntu-latest
 
+    strategy:
+      fail-fast: false
+      matrix:
+        branch: ["1.1.latest", "1.2.latest", "1.3.latest"]
```

Comment:
Same question here with determining the matrix dynamically so we don't have to manually update this with each release.  i've made an attempt to do the same (although not currently working) in another PR - https://github.com/dbt-labs/dbt-core/pull/6682

Reply:
I added a job to fetch metadata about protected branches via GitHub API: https://github.com/dbt-labs/actions/pull/68/files#diff-40a4b2214e076aa5da41caf0b207a245fb1699b11aa4c72f667c541be035a030R41-R80

Also, I updated the main job to use a dynamic matrix.

Run examples:
- https://github.com/dbt-labs/actions/actions/runs/3986791048;
- https://github.com/dbt-labs/actions/actions/runs/3986773070;